packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -190,6 +191,30 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         } catch (err: unknown) {
             PercyLogger.error(`Error while setting best platform for Percy snapshot at worker start ${err}`)
         }
+
+        // Setting up healing for those sessions where we don't add the service version capability as it indicates that the session is not being run on BrowserStack
+        if (!shouldAddServiceVersion(this._config, this._options.testObservability, caps)) {
```

Comment:
Why are we doing this in worker start? this will get called multiple times on every worker start

Reply:
It has been on the worker start since the beginning and no issues were raised in the 2 review calls as well. Do we need to move it anywhere else?

Reply:
It has been on the worker start since the beginning and no issues were raised in the 2 review calls as well. Do we need to move it anywhere else?